src/sentry/migrations/0728_add_subscription_fk_to_incident.py

```diff
@@ -0,0 +1,85 @@
+# Generated by Django 5.0.6 on 2024-06-12 16:47
+
+import logging
+
+import django.db.models.deletion
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+import sentry.db.models.fields.foreignkey
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapper
+
+logger = logging.getLogger(__name__)
+
+
+def _backfill_incident_subscription(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Incident = apps.get_model("sentry", "Incident")
+
+    # use RangeQuerySetWrapper to avoid loading all Incidents into memory
+    for incident in RangeQuerySetWrapper(Incident.objects.all()):
+        alert_rule = incident.alert_rule
+        if not alert_rule:
+            logger.info(
+                "Incident found with no alert_rule. skipping",
+                extra={"incident_id": incident.id},
+            )
+            continue
+
+        snuba_query = alert_rule.snuba_query
+        subscriptions = list(snuba_query.subscriptions.all())
+        if len(subscriptions) == 0 or len(subscriptions) > 1:
```

Comment:
Currently the only alert rules with >1 subscription would be the activated alert rules 🤔 
these subscriptions should be temporary and will expire soon.

should we backfill the incidents to just the first subscription?
This would could cause multiple `Incident`'s to be tied to the same `Subscription`...

OR

We just allow subscriptions to be  nullable on the Incident model and any _new_ Incident's will be created with a subscription